src/transformStateWithClones.js

```diff
@@ -6,8 +6,47 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const transformLogState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        transformLogState.push({ ...newState });
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing a copy of the state to `transformLogState` in every case of the switch. It's more efficient to do this just once after the switch block.